mne/filter.py

```diff
@@ -1045,9 +1046,16 @@ def create_filter(data, sfreq, l_freq, h_freq, filter_length='auto',
 
     .. versionadded:: 0.14
     """
-    sfreq = float(sfreq)
-    if sfreq < 0:
+    if sfreq is None or float(sfreq) < 0:
         raise ValueError('sfreq must be positive')
+    else:
+        sfreq = float(sfreq)
+    # If no data specified, sanity checking is invalid
+    # perform it on a dummy array that is long enough to never fail sanity
+    # checks
+    if data is None:
+        logger.info('No data specified. No sanity checks performed.')
+        data = np.empty(int(sfreq ** 2))
```

Comment:
I suspect you can actually avoid creating an array for nothing here no?

Reply:
It would be possible but it would involve changes in the `_triage_filter_params` function, which uses the data to first do:

https://github.com/mne-tools/mne-python/blob/005f4ab7fc0fc8b145172b72831b2262622b4075/mne/filter.py#L1733-L1734

and then finally do:

https://github.com/mne-tools/mne-python/blob/005f4ab7fc0fc8b145172b72831b2262622b4075/mne/filter.py#L1833-L1846

... but since `_triage_filter_params` also returns `filter_length`, I wasn't sure whether messing with it was a good idea.

Reply:
A couple of `if data is not None` conditions are fine